src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -285,3 +285,85 @@ pub fn extract_valid_column_indices(
         None => Ok(vec![]),
     }
 }
+
+/// Checks if the data type in RisingWave matches the data type in a Parquet(arrow) file.
+///
+/// This function compares the `DataType` from RisingWave with the `DataType` from
+/// Parquet file, returning `true` if they are compatible. Specifically, for `Timestamp`
+/// types, it ensures that any of the four `TimeUnit` variants from Parquet
+/// (i.e., `Second`, `Millisecond`, `Microsecond`, and `Nanosecond`) can be matched
+/// with the corresponding `Timestamp` type in RisingWave.
+pub fn is_data_type_matching(
+    rw_data_type: &risingwave_common::types::DataType,
+    arrow_data_type: &arrow_schema_iceberg::DataType,
+) -> bool {
```

Comment:
Why do we need to add a new function for the check? This leads to duplicated code, and might lead to inconsistency.

Reply:
Previously we do column pruning(aka judge the rw schema matches parquet file schame) by converting rw column to arrow `Field`, and the compare two arrow `Field`. However, the problem here is that the timestamp type of rw will only become the `Timestamp(Microsecond)` of arrow during the conversion process, this cause a mismatch between the timestamp of rw and the `Timestamp(Nanosecond)`, `Timestamp(Millisecond)`, etc. of arrow.

In short, what we need is a relatively loose type comparison.

Reply:
Do you mean 1 rw type can correspond to N arrow types?

If so, can we test type matching by converting arrow type to rw type?

Reply:
Yes, I think we can use from_field, but need to add the missing type first.
https://github.com/risingwavelabs/risingwave/blob/f559d3468a16f0a15e429dfb55276b1a07c814e3/src/common/src/array/arrow/arrow_impl.rs#L502-L514